node/core/approval-voting/src/lib.rs

```diff
@@ -2612,7 +2679,14 @@ async fn issue_approval<Context>(
 	};
 	issue_approval_span.add_int_tag("candidate_index", candidate_index as i64);
 
-	let session_info = match state.session_info(block_entry.session()) {
+	let session_info = match get_session_info(
```

Comment:
`get_sesion_info` could be a member function of `SessionInfoProvider` once we got rid of the wrapping `Option`.